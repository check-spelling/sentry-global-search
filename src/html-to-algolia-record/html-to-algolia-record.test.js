import sampleDocument from './lib/sample-document';
import htmlToAlgoliaRecord from './html-to-algolia-record';

describe('HTML to Algolia record', () => {
  let document = {};

  beforeAll(async () => {
    document = await sampleDocument();
  });

  test('meets expectations', async () => {
    const { html, title, url } = document;
    const records = await htmlToAlgoliaRecord(html, { title, url });
    expect(records).toMatchSnapshot();
  });
});
