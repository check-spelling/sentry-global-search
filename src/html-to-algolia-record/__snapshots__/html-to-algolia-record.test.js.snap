// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML to Algolia record meets expectations 1`] = `
Array [
  Object {
    "anchor": undefined,
    "objectID": "c2c149b6e273a4a57ac2ca4e49edad4c",
    "position": 1,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "For an overview of what Sentry does, take a look at the Sentry workflow.",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "2d4c55c8caf66611e2df1a7189f74b0e",
    "position": 2,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Sentry is designed to be both simple to get off the ground and powerful to grow into. If you have never used Sentry before, this tutorial helps you get started.",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "42a9ce2c5af7ba13b3aa77a1d7a11405",
    "position": 3,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Getting started with Sentry is a three step process:",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "7ffbce43cf730e1fcb7bc8d5e7b850b6",
    "position": 4,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "This will",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "0f8bb415b8b182110b81575c0e2ff080",
    "position": 5,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Test children",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "41c755f1e3eebbcbc48c91f2abf54e58",
    "position": 6,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Sign up for an account
This willTest children",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "4184ca9c8991eea4c4d90966f6b81a5c",
    "position": 7,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Install your SDK",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "b343cf9d5f0a6822bb9a81d0c9f52900",
    "position": 8,
    "section": "Getting started",
    "sectionRank": 100,
    "text": "Configure your SDK",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "461472519840ed479c4005bf5484adcc",
    "position": 9,
    "section": "Install an SDK",
    "sectionRank": 90,
    "text": "Sentry captures data by using an SDK within your application’s runtime. These are platform specific and allow Sentry to have a deep understanding of how your application works.",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "41fbc1a554f42e65d9518a9474c66b64",
    "position": 10,
    "section": "Install an SDK",
    "sectionRank": 90,
    "text": "Install our Python SDK using pip:",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "30c74743c6987e7ef26a3674dba4c0ac",
    "position": 11,
    "section": "Install an SDK",
    "sectionRank": 90,
    "text": "$ This will be indexed because it contains \`data-index\`",
    "title": "Getting started",
    "url": "https://example.com",
  },
  Object {
    "anchor": undefined,
    "objectID": "d01ffc3acf3518645f5dab5c3a371e29",
    "position": 12,
    "section": "Configure the SDK",
    "sectionRank": 80,
    "text": "After you completed setting up a project in Sentry, you’ll be given a value which we call a DSN, or Data Source Name. It looks a lot like a standard URL, but it’s actually just a representation of the configuration required by the Sentry SDKs. It consists of a few pieces, including the protocol, public key, the server address, and the project identifier.",
    "title": "Getting started",
    "url": "https://example.com",
  },
]
`;
